projects/bzip2/bzip2_decompress_target.c

```diff
@@ -0,0 +1,54 @@
+/*
+# Copyright 2018 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+###############################################################################
+*/
+
+#include "bzlib.h"
+#include <stdint.h>
+#include <stdlib.h>
+#include <assert.h>
+#include <string.h>
+
+extern int BZ2_bzBuffToBuffDecompress(char* dest,
+                                      unsigned int* destLen,
+                                      char*         source,
+                                      unsigned int  sourceLen,
+                                      int           small,
+                                      int           verbosity);
+
+int
+LLVMFuzzerTestOneInput(const uint8_t *data, size_t size)
+{
+    int r;
+    unsigned int nZ, nOut;
+
+    nOut = size*2;
+    char *outbuf = malloc(nOut);
```

Comment:
I'd highly recommend to use a static buffer for this with some fixed size (e.g. 256 KB, see zlib example: https://github.com/google/oss-fuzz/blob/master/projects/zlib/zlib_uncompress_fuzzer.cc#L11). It should give a very significant speed up.

Reply:
Maybe not. 
a static buffer might be faster, but then we may lose some asan buffer overflow checking on it. 

Also a question about `nOut = size*2;`
Does this mean that the decompression is often expected to return early due to end-of-output?